sunpy/net/scraper_utils.py

```diff
@@ -0,0 +1,108 @@
+import calendar
+from datetime import datetime, timedelta
+
+from dateutil.relativedelta import relativedelta
+
+from sunpy.extern.parse import parse
+from sunpy.time import TimeRange
+
+TIME_QUANTITIES = {'day': timedelta(days=1),
+                   'hour': timedelta(hours=1),
+                   'minute': timedelta(minutes=1),
+                   'second': timedelta(seconds=1),
+                   'millisecond': timedelta(milliseconds=1)}
+
+def check_timerange(pattern, url, timerange):
+    """
+    Checks whether the time range represented in *url* intersects
+    with the given time range.
+
+    Parameters
+    ----------
+    url : `str`
+        URL of the file.
+    timerange : `~sunpy.time.TimeRange`
+        Time interval for which files were searched.
+
+    Returns
+    -------
+    `bool`
+        `True` if URL's valid time range overlaps the given timerange, else `False`.
+    """
+    exdict = parse(pattern, url).named
+    if exdict['year'] < 100:
+        exdict['year'] = 2000 + exdict['year']
+    if 'month' not in exdict:
+                if 'month_name' in exdict:
+                    exdict['month'] = datetime.strptime(exdict['month_name'], '%B').month
+                elif 'month_name_abbr' in exdict:
+                    exdict['month'] = datetime.strptime(exdict['month_name_abbr'], '%b').month
+    tr = get_timerange_from_exdict(exdict)
+    return tr.intersects(timerange)
+
+def smallerPattern(directoryPattern):
+    """
+    Obtain the smaller time step for the given pattern.
+    """
+    try:
```

Comment:
Does this need to be in a try block?

Reply:
I don't think so, I'll revert it. I traced it back to the [second](https://github.com/sunpy/sunpy/commit/4469744d7718c97857b09d74b4ddb5a33055f7d5) scraper commit ever :P